src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/FilingInterceptor.java

```diff
@@ -1,49 +1,24 @@
 package uk.gov.companieshouse.overseasentitiesapi.interceptor;
 
-import org.springframework.lang.NonNull;
 import org.springframework.stereotype.Component;
-import org.springframework.web.servlet.HandlerInterceptor;
-import uk.gov.companieshouse.api.model.transaction.Transaction;
 import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
 
-import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.HashMap;
-import java.util.Objects;
-
-import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.ERIC_REQUEST_ID_KEY;
-import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.TRANSACTION_ID_KEY;
-import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.TRANSACTION_KEY;
-
-import static uk.gov.companieshouse.api.model.transaction.TransactionStatus.CLOSED;
 
 @Component
-public class FilingInterceptor implements HandlerInterceptor {
+public class FilingInterceptor extends AbstractClosedTransactionInterceptor {
```

Comment:
for the concrete implementation classes FilingInterceptor and ProcessingInterceptor could you add javadoc at class level just to say  eg 'should return TRUE if transaction closed' just to make it clear what the behaviour should be (for anyone reading in future)

Reply:
Sure. Didn't do this as we generally don't bother with Javadoc but makes sense to me.